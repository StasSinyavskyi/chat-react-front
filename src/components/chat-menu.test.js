/* eslint-env jest */
import renderer from 'react-test-renderer';
import React from 'react';
import ReactDOM from 'react-dom';
import ChatMenu from './chat-menu';

// if we have several test we can join them in describe
describe('<ChatMenu />', () => {
  const testParam = (
    <ChatMenu
      toChat={() => 'toChat'}
      toProfile={() => 'toProfile'}
      logout={() => 'logout'}
      isConnected={false}
    />
  );

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(testParam, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('render correctly', () => {
    const tree = renderer
      .create(testParam)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
