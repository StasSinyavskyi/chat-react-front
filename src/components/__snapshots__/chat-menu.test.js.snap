// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatMenu /> render correctly 1`] = `
<div>
  <button
    aria-haspopup="true"
    aria-owns={null}
    className="MuiButtonBase-root-39 MuiButtonBase-disabled-40 MuiButton-root-24 MuiButton-disabled-33 ChatMenu-button-23"
    disabled={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={-1}
    type="button"
  >
    <span
      className="MuiButton-label-25"
    >
      Menu
    </span>
  </button>
</div>
`;
